applications/client/src/views/Campaign/Explore/Panels/Comment/Comments.tsx

```diff
@@ -216,38 +164,20 @@ export const Comments = observer<CommentsProps>(({ sort }) => {
 				}, 250);
 			}
 		}
-	}, [data, commentsData]);
+	}, [commandGroupIdData]);
 
 	return (
 		<VirtualizedList
 			rangeChanged={(visibleRange) => state.update('visibleRange', visibleRange)}
 			listRef={listRef}
 			cy-test="comments-view"
 		>
-			{store.router.params.currentItem === 'comments_list' ? (
-				store.campaign.commentsList.commandGroupIds.length === 0 ? (
-					<MessageRow>No comments</MessageRow>
-				) : isCommentsDataLoading ? (
-					<ProgressBar intent={Intent.PRIMARY} />
-				) : (
-					store.campaign.commentsList.commandGroupIds.map((commandGroupId) => (
-						<CommentGroup
-							cy-test="comment-group"
-							key={commandGroupId}
-							commandGroupId={commandGroupId}
-							toggleNewComment={state.toggleNewComment}
-							newComment={state.newComment}
-							expandedCommandIDs={state.expandedCommandIDs}
-							removeExpandedCommandID={state.removeExpandedCommandID}
-						/>
-					))
-				)
-			) : data?.commandGroupIds.length === 0 ? (
-				<MessageRow>No comments</MessageRow>
-			) : isLoading ? (
+			{!commandGroupIdData?.commandGroupIds ? (
```

Comment:
Because we have to go through three queries to get all data, and commandGroupIdData is the first one. I would think we should have the commented out isLoading on Line 99 back to determine if we should render ProgressBar, but render ProgressBar after "No Comments" check. As below order:
1. commandGroupIdData.length===0 =>  No comments
2. isLoading (meaning commandGroupIdData.length>0 and we are still fetching their commandGroups and commands) -> ProgressBar
3. All three queried data ready => Final CommentGroup